pkg/discovery-manager/discovery_controller.go

```diff
@@ -47,13 +56,276 @@ type DiscoveryReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.2/pkg/reconcile
 func (r *DiscoveryReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	log := ctrl.LoggerFrom(ctx, "discovery", req.NamespacedName)
+	ctx = ctrl.LoggerInto(ctx, log)
 
-	// TODO(user): your logic here
+	var peeringCandidate *advertisementv1alpha1.PeeringCandidate
+
+	var discovery advertisementv1alpha1.Discovery
+	if err := r.Get(ctx, req.NamespacedName, &discovery); client.IgnoreNotFound(err) != nil {
+		klog.Errorf("Error when getting Discovery %s before reconcile: %s", req.NamespacedName, err)
+		return ctrl.Result{}, err
+	} else if err != nil {
+		klog.Infof("Discovery %s not found, probably deleted", req.NamespacedName)
+		return ctrl.Result{}, nil
+	}
+
+	klog.Infof("Discovery %s started", discovery.Name)
+
+	if discovery.Status.Phase.Phase != nodecorev1alpha1.PhaseSolved &&
+		discovery.Status.Phase.Phase != nodecorev1alpha1.PhaseTimeout &&
+		discovery.Status.Phase.Phase != nodecorev1alpha1.PhaseFailed &&
+		discovery.Status.Phase.Phase != nodecorev1alpha1.PhaseRunning &&
+		discovery.Status.Phase.Phase != nodecorev1alpha1.PhaseIdle {
+
+		setDiscoveryPhase(&discovery, nodecorev1alpha1.PhaseRunning)
+
+		if err := r.updateSolverStatus(ctx, &discovery); err != nil {
+			klog.Errorf("Error when updating Discovery %s status before reconcile: %s", req.NamespacedName, err)
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	switch discovery.Status.Phase.Phase {
+	case nodecorev1alpha1.PhaseRunning:
+		flavoursCR, err := getFlavour(discovery.Spec.Selector)
```

Comment:
This is a strange CR, also here you use the CRD defined obj ad a support variable

Reply:
Sorry, i do not understand..

Reply:
Problem fixed on the previous conversation. The problem here was that I used the `Spec`, NO more `Spec` used (`getFlavour` returns a Flavour Object)